mlflow/sklearn/__init__.py

```diff
@@ -1447,6 +1463,32 @@ def infer_model_signature(input_example):
                 estimator.predict(deepcopy(input_example)),
             )
 
+        # log datasets during training
+        if log_datasets:
+            try:
+                # create a CodeDatasetSource
+                context_tags = context_registry.resolve_tags()
+                source = CodeDatasetSource(
+                    mlflow_source_type=context_tags[MLFLOW_SOURCE_TYPE],
+                    mlflow_source_name=context_tags[MLFLOW_SOURCE_NAME],
+                )
+
+                # create a dataset
+                if isinstance(X, pd.DataFrame):
+                    dataset = from_pandas(df=X, source=source)
+                else:
```

Comment:
As with lightgbm, can we make this an `elif isinstance(X, <something else>)` with a catch-all `else` for things we don't recognize? https://github.com/mlflow/mlflow/pull/8202/files#r1173333282